app/components/ChangeLog.tsx

```diff
@@ -0,0 +1,135 @@
+import { useMemo, useState } from "react";
+import Card from "react-bootstrap/Card";
+import Table from "react-bootstrap/Table";
+import { formatDateTime } from "@/utils/formatters";
+import ChangeLogDetails from "./ChangeLogDetails";
+import {
+  ChangeLogEntry,
+  ChangeLogDiff,
+  ChangeLogEntryEnriched,
+} from "@/types/types";
+
+const KEYS_TO_IGNORE = ["updated_at", "updated_by", "position"];
+
+/**
+ * Return an array of values that are different between the `old` object and `new` object
+ * Each object in the array takes the format
+ * { field: <property name>, old: <old value>, new: <new value> }
+ */
+const getDiffArray = <T extends Record<string, unknown>>(
+  old: T,
+  new_: T
+): ChangeLogDiff[] => {
+  const diffArray = Object.keys(new_).reduce<ChangeLogDiff[]>((diffs, key) => {
+    if (new_[key] !== old[key] && !KEYS_TO_IGNORE.includes(key)) {
+      diffs.push({
+        field: key,
+        old: old[key],
+        new: new_[key],
+      });
+    }
+    return diffs;
+  }, []);
+
+  // Sort array of entries by field name
+  diffArray.sort((a, b) =>
+    a.field < b.field ? -1 : a.field > b.field ? 1 : 0
+  );
+
+  return diffArray;
+};
+/**
+ * Prettify the user name if it is `cris`
+ */
+const formatUserName = (userName: string): string =>
+  userName === "cris" ? "TxDOT CRIS" : userName;
```

Comment:
i just want to point out that this function has a return type defined (`string`). typescript will infer the return type based on what your code returns, but as a best practice i think we want to strive for explicitly naming the return type. 

there are some spots where i failed to do this and i'm keeping an eye out for them.